actor/v7action/service_instance_sharing.go

```diff
@@ -18,9 +18,48 @@ func (actor Actor) ShareServiceInstanceToSpaceAndOrg(
 ) (Warnings, error) {
 	var serviceInstance resources.ServiceInstance
 	var shareSpace resources.Space
-	var shareToOrgGUID = targetedOrgGUID
 
 	return handleServiceInstanceErrors(railway.Sequentially(
+		func() (warnings ccv3.Warnings, err error) {
+			var actorWarnings Warnings
+			serviceInstance, shareSpace, actorWarnings, err = actor.validateSharingDetails(serviceInstanceName, targetedSpaceGUID, targetedOrgGUID, sharedToDetails)
+			warnings = ccv3.Warnings(actorWarnings)
+			return
+		},
+		func() (warnings ccv3.Warnings, err error) {
+			_, warnings, err = actor.CloudControllerClient.ShareServiceInstanceToSpaces(serviceInstance.GUID, []string{shareSpace.GUID})
+			return
+		},
+	))
+
+	return Warnings{}, nil
+}
+
+func (actor Actor) UnshareServiceInstanceFromSpaceAndOrg(
+	serviceInstanceName, targetedSpaceGUID, targetedOrgGUID string,
+	sharedToDetails ServiceInstanceSharingParams,
+) (Warnings, error) {
+	return handleServiceInstanceErrors(railway.Sequentially(
+		func() (warnings ccv3.Warnings, err error) {
+			var actorWarnings Warnings
+			_, _, actorWarnings, err = actor.validateSharingDetails(serviceInstanceName, targetedSpaceGUID, targetedOrgGUID, sharedToDetails)
+			warnings = ccv3.Warnings(actorWarnings)
+			return
+		},
+	))
+
+	return Warnings{}, nil
+}
+
+func (actor Actor) validateSharingDetails(
+	serviceInstanceName, targetedSpaceGUID, targetedOrgGUID string,
+	sharedToDetails ServiceInstanceSharingParams,
+) (resources.ServiceInstance, resources.Space, Warnings, error) {
+	var serviceInstance resources.ServiceInstance
+	var shareSpace resources.Space
+	var shareToOrgGUID = targetedOrgGUID
```

Comment:
Is this variable used at all?

Reply:
Yes, further down in the call to GetSpaceByNameAndOrganization. targetedOrgGUID is the one from were the command was ran, but then it can be overriden by a flag, so shareToOrgGUID will have the value we need to send